### PR TITLE
DR-638 Michael's big table refactor

### PR DESCRIPTION
**Pull request by myessail**

### CODE REVIEW

File: src/components/dataset/query/QueryView.jsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export class QueryView extends React.PureComponent {
         runQuery(
           dataset.dataProject,
           `#standardSQL
-          SELECT * EXCEPT (${DB_COLUMNS.ROW_ID}) FROM \`${dataset.dataProject}.datarepo_${dataset.name}.${selected}\`
```
Comment (myessail) [inline src/components/dataset/query/QueryView.jsx:60]: using this to assign the `key` property to each row in the table, but still hiding it from the user

File: src/components/table/LightTable.jsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  },
+  searchInput: {
+    paddingTop: theme.spacing(1.5),
+    paddingLeft: theme.spacing(6),
+  },
+});
+
+const DEFAULT_PAGE_SIZE = 10;
+
+export class LightTable extends React.PureComponent {
```
Comment (myessail) [inline src/components/table/LightTable.jsx:67]: previously JadeTable

File: package.json [hunk 3]
```diff
@@ -44,7 +44,6 @@
     "jest-cli": "^24.9.0",
     "js-combinatorics": "^0.5.4",
     "lodash": "^4.17.15",
-    "material-table": "^1.54.1",
```
Comment (thathert) [inline package.json:47]: Yayy

File: src/components/table/JadeTable.jsx [hunk 4]
```diff
@@ -1,207 +1,171 @@
...
 
   handleChangeRowsPerPage = event => {
-    const { handleEnumeration } = this.props;
-    const { orderDirection, orderBy, page, searchString } = this.state;
-    const limit = event.target.value;
-    const offset = page * limit;
-    this.setState({ rowsPerPage: limit });
-    handleEnumeration(limit, offset, orderBy, orderDirection, searchString);
+    this.setState({
+      rowsPerPage: +event.target.value,
```
Comment (thathert) [inline src/components/table/JadeTable.jsx]: I used to do always do this, but found it's confusing to readers unfamiliar with the quirks of js. Better to do the less-cool-but-more-readable `parseInt`

File: src/components/table/JadeTable.jsx [hunk 5]
```diff
@@ -1,207 +1,171 @@
...
+              <JadeTableHead
+                columns={columns}
+                orderBy={orderBy}
+                order={order}
+                createSortHandler={this.createSortHandler}
+              />
+              <TableBody>
+                {rows.map(row => {
+                  return (
+                    <TableRow hover tabIndex={-1} key={row.code}>
```
Comment (thathert) [inline src/components/table/JadeTable.jsx]: Why tabIndex -1

File: src/components/table/JadeTableHead.jsx [hunk 6]
```diff
@@ -1,70 +1,51 @@
...
+                  align={column.numeric ? 'right' : 'left'}
+                  padding={column.disablePadding ? 'none' : 'default'}
+                  sortDirection={orderBy === column.id ? order : false}
                 >
                   <TableSortLabel
-                    active={orderBy === col.property}
-                    direction={orderDirection}
-                    onClick={this.createSortHandler(col.property)}
+                    active={orderBy === column.id}
+                    direction={order !== '' ? order : 'desc'}
```
Comment (thathert) [inline src/components/table/JadeTableHead.jsx:42]: What about if order is null or undefined?

File: src/modules/bigquery.js [hunk 7]
```diff
@@ -6,101 +6,56 @@ export default class BigQuery {
...
 
-                if (currColumn.datatype === 'integer') {
-                  item = this.commaFormatted(item);
-                }
+  createData = (columns, row) => {
+    let i = 0;
+    const res = {};
+    for (i = 0; i < columns.length; i++) {
+      const column = columns[i];
+      const columnId = _.get(column, 'id');
```
Comment (thathert) [inline src/modules/bigquery.js]: What does this gain us over using `column.id`? I can see usefulness in the cases where the properties are nested inside of the object, but this seems unnecessary.

File: src/modules/bigquery.js [hunk 8]
```diff
@@ -6,101 +6,56 @@ export default class BigQuery {
...
 
-          resolve({
-            data,
-            page: query.page,
-            totalCount: parseInt(queryResults.totalRows, 10),
-          });
-        });
-    });
+      if (columnId === 'datarepo_row_id') {
+        res.code = value;
```
Comment (thathert) [inline src/modules/bigquery.js]: Why `code`?

**Review by thathert — CHANGES_REQUESTED** [review]
🤶

**Review by RoriCremer — APPROVED** [review]

**Review by thathert — APPROVED** [review]
👍 LGTM

**Pull request merged**